### ISSUE TITLE
C3DTexEnvOp() regarding alpha channel - writing into wrong spots

### ISSUE THREAD

**Issue by DiscostewSM** [fincs/citro3d#43]
Based on the source code for the texture environment structures, you have RGB and Alpha separated. That's correct for most parts, but the operand has a slight overlap and a 1-bit separation with Alpha. In sections of 4-byte word (0:31), RGB operand components R, G, and B cover bits 0:3, 4:7, and 8:11. The Alpha operand structure does not follow this, but leaks into the previous first half of the word type, starting with 12:14 for the first component, then skips a bit to set 16:18 for the 2nd component, then another bit skip for 20:22 with the 3rd component. Found this out when attempting to use ONE_MINUS_SRC_ALPHA with the first component and nothing changed. Manually added 0x1000 to opRGB after setting up the overall texture combiner, and it worked in flipping the alpha values of a texture.

**Comment by fincs** [comment]
Yeah, this is a known issue in upstream citro3d. I have a development branch in the works which fixes this issue. Since the RGB/Alpha OP enums are incompatible, I decided to remove C3D_TexEnvOp() altogether, and replace it with C3D_TexEnvOpRgb and C3D_TexEnvOpAlpha().

**Comment by DiscostewSM** [comment]
Gave it some thought last night in bed, but couldn't you just alter the C3D_TexEnv structure slightly? Instead of the current "u16 opRgb, opAlpha", make a union and size them up?

```
union
	{
		u32 opRgbA;
		struct
		{
			u32 opRgb	: 12;
			u32 opAlpha	: 12;
			u32 opEmpty	: 8;
		};
	};
```

While the alpha only uses 3 bits per component, the offset from the beginning of one component to the next is 4 bits, the same size as for RGB, allowing for the existing RGB/Alpha OP enums via GPU_TEVOPERANDS to work with it. Did a test myself with this change, and I don't seem to run into any problems.

**Comment by fincs** [comment]
That's in part what I did. However the values of the enums themselves are incompatible - it doesn't make sense to use C3D_Both.

**Comment by DiscostewSM** [comment]
True, since the only time C3D_Both would even work properly is when using the default 0 for it all.
